c7n/resources/asg.py

```diff
@@ -1507,10 +1507,28 @@ def process_asg(self, asg):
                 return
             raise
         ec2_client = session.client('ec2')
+        asg_instance_ids = [i['InstanceId'] for i in asg['Instances']]
+        if not asg_instance_ids:
+            log.info("No instances in asg to stop %s" % (
+                asg['AutoScalingGroupName']))
+            return
+
+        # Determine which EC2s in ASG can be stopped:
+        # - Only EBS-backed EC2s, No Spot instances, Only running instances.
+        response = ec2_client.describe_instances(InstanceIds=asg_instance_ids)
```

Comment:
it seems like we should be using the ec2 resource manager here to pick these up for the full set of resources, that will give us some utility out of the cache/retry on describe etc.

given the duplication between start/stop to do this, it might be good for a base class method that they can both call, and just parameterize the instance state.

Reply:
Thanks for the clarification on the CLA...I did just submit the individual too.

I'm not quite sure that I understand the ec2 resource manager comment...is there an example of doing this?  Good feedback on the duplication comment--I can change that...

Reply:
re resource manager, nutshell is that if you fetch the instances with the following it has some other nice properties (will do retry on rate limit, can utilize cache, will respect describe source (ie if using config)).

```python
instances = self.manager.get_resource_manager('ec2').get_resources(instance_ids)
```

Reply:
Thanks, just updated.